client/my-sites/themes/install-theme-button.tsx

```diff
@@ -25,7 +34,7 @@ const InstallThemeButton = ( {
 		return null;
 	}
 
-	let siteType = null;
+	let siteType: string | null = null;
```

Comment:
while it's good indeed to limit the scope of type changes, sometimes little utilities like this can be extracted into a separate function and that lets us avoid the `null` in the type _and_ the `let`. it has the potential to clarify the logic and also that `siteType` won't be changing again down below in the function. I only write this to point it out as something I like to do judiciously when converting from JavaScript to TypeScript.

```ts
type SiteType = 'logged_out' | 'atomic' | 'jetpack' | 'simple';

const inferSiteType = ( {
	isLoggedIn,
	atomicSite,
	jetpackSite,
	siteSlug
}: InstallThemeButtonProps ): SiteType => {
    if ( ! isLoggedIn ) {
		return 'logged_out';
	}

	…
}

…

const InstallThemeButton = … => {
	…

	const siteType = inferSiteType( props );

	…
}
```